scripts/run_integration_tests.sh

```diff
@@ -5,23 +5,48 @@ function get_frequency_pid () {
 }
 
 function cleanup () {
-    if ${SHOULD_KILL}
-    then
-        ${RUNDIR}/kill_freq.sh
-    fi
+    local signal="$1"
+
+    case "$signal" in
+        TERM|INT) 
+            # Catch TERM and INT signals and exit gracefully
+            exit
+            ;;
+        EXIT)
+            if [ "${SHOULD_KILL}" = true ]
+            then
+                # kill_freq.sh is not used here because we do not know what directory
+                # the script is in when a signal is received. Therefore, we do not
+                # know how to navigate to the kill_freq.sh script with relative paths.
+                PID=$( get_frequency_pid )
+                if [ -n "${PID}" ]
+                then
+                    echo "Killing Frequency... PID: ${PID}"
+                    kill -9 ${PID}
+                    echo "Frequency has been killed. 💀"
+                else
+                    echo "Frequency is already gone. 💀"
+                fi 
+            fi
+            ;;
+    esac
 }
 
 RUNDIR=$(dirname ${0})
 SKIP_JS_BUILD=
-trap cleanup EXIT KILL INT
+trap 'cleanup EXIT' EXIT
+trap 'cleanup TERM' TERM
+trap 'cleanup INT' INT
 
-while getopts "s" OPTNAME
+while getopts "s:" OPTNAME
 do
-    case ${OPTNAME} in
-        "s") SKIP_JS_BUILD=1
+    case "${OPTNAME}" in
+        "s") SKIP_JS_BUILD=$OPTARG
```

Comment:
Why the change to use `$OPTARG` here? Seems like it just puts more burden on the user to pass a meaningless value on the CLI.

Reply:
Good point, reverting to the original.